client/js/components/snippets/snippet.jsx

```diff
@@ -56,7 +78,47 @@ export default class Snippet extends React.Component {
     }
   }
 
-  render() {
+  _getEditableSnippet() {
```

Comment:
I think in the future we should split it into separate components because it starts to look like a mess